src/integrations/stripe.ts

```diff
@@ -162,6 +143,86 @@ export async function setupPaymentMethodAndPay(
   return null;
 }
 
+/**
+ * Used to either collect payment directly or launch stripe checkout to collect payment.
+ */
+export async function collectPayment(
+  actionData: ActionData,
+  stripePromise = getStripe()
```

Comment:
not sure it's best practice to have a function call be the default. maybe default should be null or something and call getStripe() explicitly in the function?